static/src/javascripts/projects/commercial/modules/dfp/render-advert-label.js

```diff
@@ -5,14 +5,29 @@ import popupTemplate from 'raw-loader!commercial/views/ad-feedback-popup.html';
 import tick from 'svgs/icon/tick.svg';
 import { commercialFeatures } from 'commercial/modules/commercial-features';
 
-const shouldRenderLabel = adSlotNode =>
-    !(
-        adSlotNode.classList.contains('ad-slot--fluid') ||
-        adSlotNode.classList.contains('ad-slot--frame') ||
-        adSlotNode.classList.contains('gu-style') ||
-        adSlotNode.getAttribute('data-label') === 'false' ||
-        adSlotNode.getElementsByClassName('ad-slot__label').length
+/* creatives can explicitly request to have - or to not have - an 'Advertisment' label added to them
+ by containing a div with the following class followed by true/false.
+ */
+const labelRequiredSuffix = 'js-advertisement-label-required';
+
+const shouldRenderLabel = adSlotNode => {
+    const explicitlyOptedOut =
+        adSlotNode.getElementsByClassName(`${labelRequiredSuffix}-false`)
+            .length > 0;
+    const explicitlyOptedIn =
+        adSlotNode.getElementsByClassName(`${labelRequiredSuffix}-true`)
+            .length > 0;
+
+    return (
+        (!explicitlyOptedOut || explicitlyOptedIn) &&
```

Comment:
Might be worth assigning this expression to a well named variable and returning that for readability.

Reply:
Yeah - I'm planning on ripping this all out in a day or so - the logic is horrendous at the moment.